AK/LEB128.h

```diff
@@ -13,26 +13,19 @@
 namespace AK {
 
 struct LEB128 {
-    template<typename StreamT, typename ValueType = size_t>
-    static bool read_unsigned(StreamT& stream, ValueType& result)
+    template<typename ValueType = size_t>
+    static bool read_unsigned(DeprecatedInputStream& stream, ValueType& result)
     {
-        [[maybe_unused]] size_t backup_offset = 0;
-        if constexpr (requires { stream.offset(); })
-            backup_offset = stream.offset();
```

Comment:
How confident are we that neither LibWasm nor LibDebug were relying on this backup behavior of the LEB128 reader?

Reply:
Quite confident, since neither do we continue parsing in case any error was encountered at the moment, nor do any Streams that we are passing in here (as of `HEAD`) support `offset()` and/or `seek()` anyways, since they all have been wrapped in `WrapInAKInputStream`.